### ISSUE TITLE
Project Proposal Feedback

### ISSUE THREAD

**Issue by poojapathak1725** [COGS118A/Group046-Wi23#9]
~Project Proposal Grade: 3.5/9~
Updated Project Proposal Grade - 7.75/9

Please delete the template instructions provided in the notebook. 

Title and Abstract (1/1)
Please include an informative title for your project.

Background (0/1)
Did not have at least three sources. 
Please include more detailed documentation of the usage of Hamming Distance and its relevance. Why do you want to do this project? Who and how will this help? Have there been any prior works conducted in this space, and what models did they use? What did they learn? How is your project different? Address these points. (-1)

Research Problem Statement (& Significance, Purpose of Project) (0/1.5) 
Missing/Incomplete (-1.5)

Data - (0/1.5)
First clearly state the number of observations and features in bold. Add a link to your dataset/source of your dataset. 
Describe the critical features that you believe are relevant for approximating distances - are they numeric/categorical? Range? 
Then include a short section on data pre-processing/transformation steps needed - dropping missing rows, feature imputation techniques, one-hot-encoding for eg. 
If you are constructing your training data - how many observations do you need? What do you want your training data to look like? Include details (-1.5)

Proposed Solution (0.5/1.5)- 
Missing model selection/cross-validation to test the generalization of models. For this section, we expect to see a compare/contrast of at least 3 models and their performance on metrics to evaluate generalization performance. Additionally, reason for selecting this model is missing - Why do you think the multi-layered perception will be suitable for this task? What about other approaches? How do you plan to measure this time for the forward pass? 
Is the bk tree your baseline model? This section needs to be rewritten so the steps to your proposed solution are clearer. (-1)

Metrics - (1/1.5) 
We need more than one evaluation metric for this section. How did you decide on the value 75%? We can't arbitrarily choose this number - and if you do, back it up with sources that use the same. (-0.5)

Ethics and Privacy (0.5/0.5) - 
OK

Team Expectations (0.25/0.25)
- OK

Timeline (0.25/0.25)
- OK

Other Comments -

For a project within scope of this class, we expect to see at a high level - 
- Training multiple supervised learning models on the same dataset (using an appropriate train/test split)
- Tuning hyperparameters using grid search/random search +  Cross-validation 
- Evaluating the performance of these models based on your metrics 
- Discussion around why some models performed better than others

I feel like your project proposal is not quite within the scope/not what we expect to see. 
You will likely also spend time constructing the training data/test data for this project - which isn't really the main point here. 
If you'd like to stick with this topic - please incorporate the feedback given and clarify your proposed solution as a reply here/in the upcoming checkpoint. Otherwise, I would suggest finding an alternate dataset and refining your question. 

You can reply to this feedback below. Make sure to incorporate this feedback so you can make up these points at the checkpoint. Contact me anytime if you want help improving your project or have any questions at all!
